### PR TITLE
Issues #480 #617. Alerter doesn't recognise alert.type from  global.properties (only from UI), refactoring of Alert notification code

### PR DESCRIPTION
**Pull request by antey86**
The fix for

- [Alerter doesn't recognise global.properties for alert.type #480](https://github.com/azkaban/azkaban/issues/480)

- ["alert.type" property](https://github.com/azkaban/azkaban/issues/617)

1. Custom alerters can not be set for all Azkaban Flows in global.properties or even uploaded flow properties files.
alert.type is specifically checked from the UI flow parameters only. It should check all possible configuration locations.

2. Refactoring of Email/Alert notification code has been performed

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java [hunk 1]
```diff
@@ -154,6 +159,28 @@ public ExecutorManager(final Props azkProps, final ExecutorLoader loader,
...
+  }
+
+  private Props setupGlobalProps(Props azkProps) {
+    String globalPropsPath = azkProps.getString("executor.global.properties", null);
+    if (Objects.nonNull(globalPropsPath)) {
+      try {
+        if (globalPropsPath.startsWith("/")) {
+          return new Props(null, new File(globalPropsPath));
+        } else {
+          final URL globalPropsUrl = Resources.getResource(globalPropsPath);
```
Comment (reallocf) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java]: Do we need to introduce the new `Resources` dependency? Can we accomplish the same thing with the the Java `File` library?

File: azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java [hunk 2]
```diff
@@ -154,6 +159,28 @@ public ExecutorManager(final Props azkProps, final ExecutorLoader loader,
     this.cleanerThread = new CleanerThread(executionLogsRetentionMs);
     this.cleanerThread.start();
 
+    this.globalProps = setupGlobalProps(azkProps);
+  }
+
+  private Props setupGlobalProps(Props azkProps) {
+    String globalPropsPath = azkProps.getString("executor.global.properties", null);
+    if (Objects.nonNull(globalPropsPath)) {
```
Comment (reallocf) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java]: `Objects.nonNull` isn't a typical way that we check if a variable is null.

Maybe we can just use `globalPropsPath != null`?

File: azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java [hunk 3]
```diff
@@ -1262,60 +1289,52 @@ private void finalizeFlows(final ExecutableFlow flow) {
     // the reference.
 
     this.updaterStage = "finalizing flow " + execId + " alerting and emailing";
-    if (alertUser) {
-      final ExecutionOptions options = flow.getExecutionOptions();
-      // But we can definitely email them.
+    if (alertUser) sendAlertNotification(flow);
```
Comment (reallocf) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java]: Can we include braces here?

So something like:
```
if (alertUser) {
    sendAlertNotification(flow);
}
```

**Comment by antey86** [comment]
fixed

**Comment by antey86** [comment]
All notes were fixed, any reason to close this PR?

**Comment by antey86** [comment]
@burgerkingeater please review

**Pull request closed without merging**